data/crac/crac-io/crac-io-cim/src/main/java/com/powsybl/openrao/data/crac/io/cim/craccreator/HvdcRangeActionCreator.java

```diff
@@ -111,6 +99,30 @@ public void addDirection(RemedialActionSeries remedialActionSeries) {
         }
     }
 
+    private Boolean readRangeAndCheckIfInverted(Boolean isRemedialActionSeriesInverted, RemedialActionRegisteredResource registeredResource, String networkElementId) {
+        boolean isRegisteredResourceInverted = readHvdcRange(
+            networkElementId,
+            registeredResource.getResourceCapacityMinimumCapacity().intValue(),
+            registeredResource.getResourceCapacityMaximumCapacity().intValue(),
+            registeredResource.getInAggregateNodeMRID().getValue(),
+            registeredResource.getOutAggregateNodeMRID().getValue());
+
+        if (Objects.nonNull(isRemedialActionSeriesInverted) && !isRemedialActionSeriesInverted.equals(isRegisteredResourceInverted)) {
+            throw new OpenRaoImportException(ImportStatus.INCONSISTENCY_IN_DATA, "HVDC registered resources reference lines in opposite directions");
+        } else {
+            return isRegisteredResourceInverted;
+        }
+    }
+
+    private void checkHvdcNetworkElementAndInitAdder(RemedialActionRegisteredResource registeredResource, String networkElementId) {
+        checkHvdcNetworkElement(networkElementId);
+        HvdcLine hvdcLine = network.getHvdcLine(networkElementId);
+
+        if (hvdcLine.getConverterStation1().getTerminal().isConnected() && hvdcLine.getConverterStation2().getTerminal().isConnected()) {
+            hvdcRangeActionAdders.putIfAbsent(networkElementId, initHvdcRangeActionAdder(registeredResource));
+        }
```

Comment:
Should we add some logs in the else part here, to explain why RA was not imported ?